### PR TITLE
Use `optimize_module` pass for the quantization to fp16

### PR DESCRIPTION
**Pull request by umangyadav**
Fixes https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/issues/1746

BatchNorm only has `x` as the runtime input parameter for the following equation. All the other parameters are compile-time constants and related operations can be const-folded before quantizing to fp16 to preserve precision. 

`1/sqrt((var+epsilon)) * gamma` can be const folded. 

Adding `optimize_module()` pass before `quantize_fp16()` would do the simplification and const-folding. 

image

This PR along with #1969 produces FP16 ResNet50 accuracy same as Fp32 model ~76.45%

### CODE REVIEW

File: src/onnx/parse_batchnorm.cpp
Comment (kahmed10) [inline src/onnx/parse_batchnorm.cpp:1]: If optimize_module is all that's needed, why do the parsers need to be modified?

Comment (umangyadav) [reply src/onnx/parse_batchnorm.cpp:1]: parser changes the order of computation. 

It calculuates `1/sqrt((var+epsilon)) * gamma` first.

File: src/pass_manager.cpp [hunk 1]
```diff
@@ -178,6 +178,8 @@ void run_passes(module& mod, const std::vector<pass>& passes, tracer trace)
 
 void run_passes(program& prog, const std::vector<pass>& passes, tracer trace)
 {
+    if(enabled(MIGRAPHX_TRACE_PASSES{}))
```
Comment (kahmed10) [inline src/pass_manager.cpp:181]: Is this needed for the PR?

Comment (umangyadav) [reply src/pass_manager.cpp:181]: not needed actually. I can remove it.

**Review by kahmed10 — APPROVED** [review]

**Review by turneram — APPROVED** [review]

**Review by shivadbhavsar — APPROVED** [review]

**Pull request merged**